### PR TITLE
HW13-Tests

### PR DESCRIPTION
**Pull request by my20yeezy**

### CODE REVIEW

File: src/test/java/com/ernie/app/ticket/TicketServiceTest.java [hunk 1]
```diff
@@ -1,132 +1,251 @@
...
+    private TicketRepository ticketRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private TicketServiceImpl ticketService;
+
+    @Value("${user_update_and_ticket_creation.enabled}")
+    public boolean isUserUpdateAndTicketCreationEnabled;
```
Comment (AtticSiol) [inline src/test/java/com/ernie/app/ticket/TicketServiceTest.java]: in test context this value will not be initialized by value from properties, you can remove it

File: src/test/java/com/ernie/app/ticket/TicketServiceTest.java [hunk 2]
```diff
@@ -1,132 +1,251 @@
...
+    void saveTicket_negativeTest_creationDisabled() {
+        ticketService.isUserUpdateAndTicketCreationEnabled = false;
+
+        assertThrows(IllegalArgumentException.class, () -> ticketService.saveTicket(ticket));
+        verify(ticketRepository, never()).save(ticket);
+    }
+    @Test
+    void saveTicket_cornerCase_nullTicket() {
+        assertThrows(NullPointerException.class, () -> ticketService.saveTicket(null));
+        verify(ticketRepository, never()).save(any());
```
Comment (AtticSiol) [inline src/test/java/com/ernie/app/ticket/TicketServiceTest.java]: do not use `any()`, specify what exactly you expected had to be saved

File: src/test/java/com/ernie/app/ticket/TicketServiceTest.java [hunk 3]
```diff
@@ -1,132 +1,251 @@
...
+        assertNotNull(foundTicket);
+        assertEquals(ticketId, foundTicket.getId());
+        verify(ticketRepository, times(1)).getReferenceById(ticketId);
+    }
+    @Test
+    void getTicketById_negativeTest_nonExistentId() {
+        UUID nonExistentId = UUID.randomUUID();
+        when(ticketRepository.getReferenceById(nonExistentId)).thenReturn(null);
+
+        Ticket result = ticketService.getTicketById(nonExistentId);
```
Comment (AtticSiol) [inline src/test/java/com/ernie/app/ticket/TicketServiceTest.java:91]: add `assertEquals` to compare expected and actual values

File: src/test/java/com/ernie/app/ticket/TicketServiceTest.java [hunk 4]
```diff
@@ -1,132 +1,251 @@
...
+        assertThrows(IllegalArgumentException.class, () -> ticketService.getTicketById(null));
+        verify(ticketRepository, never()).getReferenceById(any());
+    }
+
+
+    @Test
+    void getAllTickets_positiveTest() {
+        when(ticketRepository.findAll()).thenReturn(List.of(ticket));
+
+        List<Ticket> tickets = ticketService.getAllTickets();
```
Comment (AtticSiol) [inline src/test/java/com/ernie/app/ticket/TicketServiceTest.java:108]: every time when you return something you need to compare this result with some expected result

**Comment by my20yeezy** [comment]
@AtticSiol fixed all in https://github.com/my20yeezy/Learning-at-Andersen/pull/13/commits/8a07d68c0681b5fa6c124eecf4c398669ccbd3a1

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**